jvm-runtime/ftl-runtime/common/deployment/src/main/java/xyz/block/ftl/deployment/ModuleProcessor.java

```diff
@@ -0,0 +1,67 @@
+package xyz.block.ftl.deployment;
+
+import java.io.IOException;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.util.Objects;
+import java.util.stream.Collectors;
+
+import org.jboss.logging.Logger;
+import org.tomlj.Toml;
+import org.tomlj.TomlParseResult;
+
+import io.quarkus.deployment.annotations.BuildStep;
+import io.quarkus.deployment.builditem.ApplicationArchivesBuildItem;
+import io.quarkus.deployment.builditem.ApplicationInfoBuildItem;
+import io.quarkus.deployment.builditem.FeatureBuildItem;
+import io.quarkus.deployment.builditem.RunTimeConfigBuilderBuildItem;
+import xyz.block.ftl.runtime.config.FTLConfigSourceFactoryBuilder;
+
+public class ModuleProcessor {
+
+    private static final Logger log = Logger.getLogger(ModuleProcessor.class);
+
+    private static final String FEATURE = "ftl-java-runtime";
+
+    @BuildStep
+    ModuleNameBuildItem moduleName(ApplicationInfoBuildItem applicationInfoBuildItem,
```

Comment:
Aaaah, so this is done as part of the build. I was wondering how that would work. Unfortunately we can't do this in Go, thus the `ftl/echo` namespaces.